src/wallet/rpcwallet.cpp

```diff
@@ -687,6 +687,8 @@ UniValue getbalance(const JSONRPCRequest& request)
     if (request.params.size() == 0)
         return  ValueFromAmount(pwalletMain->GetBalance());
 
+    const string* account = request.params[0].get_str() != "*" ? &request.params[0].get_str() : nullptr;
```

Comment:
Yes, this is just keeping the current behavior (line 698 before this commit, which is removed in the next commit).